crates/ruma-events/tests/it/beacon.rs

```diff
@@ -0,0 +1,48 @@
+#![cfg(feature = "unstable-msc3489")]
+
+use js_int::uint;
+use ruma_common::{owned_event_id, MilliSecondsSinceUnixEpoch};
+use ruma_events::beacon::BeaconEventContent;
+use serde_json::{from_value as from_json_value, json, Value as JsonValue};
+
+fn get_beacon_event_content() -> BeaconEventContent {
+    BeaconEventContent::new(
+        owned_event_id!("$beacon_info_event_id:example.com"),
+        "geo:51.5008,0.1247;u=35".to_owned(),
+        Some(MilliSecondsSinceUnixEpoch(uint!(1_636_829_458))),
+    )
+}
+
+fn get_beacon_event_content_json() -> JsonValue {
+    json!({
+        "m.relates_to": {
+            "rel_type": "m.reference",
+            "event_id": "$beacon_info_event_id:example.com"
+        },
+        "org.matrix.msc3488.location": {
+            "uri": "geo:51.5008,0.1247;u=35",
+        },
+        "org.matrix.msc3488.ts": 1_636_829_458
+    })
+}
+
+#[test]
+fn beacon_event_content_serialization() {
```

Comment:
Could we have tests of full event (`BeaconEvent`) serialization and deserialization too?

Reply:
something like this?

```
#[test]
fn beacon_full_event_deserialization() {
    let json_data = json!({
        "content": get_beacon_event_content_json(),
        "event_id": "$beacon_event_id:example.com",
        "origin_server_ts": 1_636_829_458,
        "room_id": "!roomid:example.com",
        "type": "org.matrix.msc3672.beacon",
        "sender": "@example:example.com"
    });

    let event = from_json_value::<AnyMessageLikeEvent>(json_data).unwrap();

    assert_matches!(event, AnyMessageLikeEvent::Beacon(MessageLikeEvent::Original(message_event)));
    assert_eq!(
        message_event.content.relates_to.event_id,
        owned_event_id!("$beacon_info_event_id:example.com")
    );
    assert_matches!(message_event.content.relates_to, Reference { event_id, .. });
}
```

Reply:
Yes, that would be perfect.

Reply:
Could you point me to a good example for serialization? Deserialize is complete

Reply:
Oh sorry, I forgot for a minute that it's not possible to serialize whole events. We'll be fine with just deserialization.